lib/psych/visitors/yaml_tree.rb

```diff
@@ -310,19 +310,20 @@ def visit_String o
           style = Nodes::Scalar::LITERAL
           plain = false
           quote = false
-        elsif o =~ /\n/
+        elsif o =~ /\n[^\Z]/  # match \n except blank line at the end of string
           style = Nodes::Scalar::LITERAL
         elsif o == '<<'
           style = Nodes::Scalar::SINGLE_QUOTED
           tag   = 'tag:yaml.org,2002:str'
           plain = false
           quote = false
+        elsif o.length > line_width
+          style = Nodes::Scalar::FOLDED
+          o << "\n" unless o =~ /\n\Z/  # to avoid non-default chomping indicator
```

Comment:
This doesn’t work, it breaks `TestString#test_folded_when_longer_than_line_width_and_no_newlines`. The problem is with [this line](https://github.com/jirutka/psych/blob/patch-2/lib/psych/visitors/yaml_tree.rb#L304).

When I assign it to `str` instead of `o`, then tests passes, but I’m not sure if it’s really okay. Why is `o` assigned to `str` and then both `o` and `str` are used later in this method?